internal/controller/management_controller.go

```diff
@@ -211,6 +218,47 @@ func (r *ManagementReconciler) Update(ctx context.Context, management *hmc.Manag
 	return ctrl.Result{}, nil
 }
 
+func (r *ManagementReconciler) startDependentControllers(ctx context.Context, management *hmc.Management) error {
+	l := ctrl.LoggerFrom(ctx)
+
+	if !r.SveltosDependentControllersStarted {
+		if management.Status.Components[hmc.ProviderSveltosName].Success {
+			dc, err := dynamic.NewForConfig(r.Manager.GetConfig())
+			if err != nil {
+				return fmt.Errorf("failed to create dynamic client: %w", err)
+			}
+
+			currentNamespace := utils.CurrentNamespace()
+
+			l.Info(fmt.Sprintf("Provider %s has been successfully installed, so setting up controller for ClusterDeployment", hmc.ProviderSveltosName))
+			if err = (&ClusterDeploymentReconciler{
+				Client:          r.Manager.GetClient(),
+				Config:          r.Manager.GetConfig(),
+				DynamicClient:   dc,
+				SystemNamespace: currentNamespace,
+			}).SetupWithManager(r.Manager); err != nil {
+				return fmt.Errorf("failed to setup controller for ClusterDeployment: %w", err)
```

Comment:
i think we should panic in that case, wdyt? the same comment for the second ctrl